qemu/tests/check_coredump.py

```diff
@@ -148,51 +147,76 @@ def run(test, params, env):
     4) If yes, copy them to working directory.
     """
 
+    (crash_dirs, msgbox) = ([], [])
+    error.context("Check process crash file in test result dir", logging.info)
+    root_dir = os.path.dirname(os.path.dirname(test.debugdir))
+    for prefix in ["crash", "core"]:
+        pattern = "%s/*/%s.*" % (root_dir, prefix)
+        crash_dirs += glob.glob(pattern)
+    if crash_dirs:
+        crash_files = map(lambda x: "/".join(x.split("/")[-2:]), crash_dirs)
+
     # Preliminary
     # yum install libguestfs libguestfs-tools libguestfs-winsupport
     try:
         os_dep.command("guestmount")
-    except:
-        warn_msg = "Need packages: libguestfs libguestfs-tools" + \
-                   " libguestfs-winsupport"
+        # define the file name need to be checked
+        file_check_win_default = "Windows/dump"
+        file_check_linux_default = "var/crash/*"
+        host_mountpoint_default = "mnt/mountpoint"
+
+        host_mountpoint = params.get(
+            "host_mountpoint",
+            host_mountpoint_default)
+        host_mountpoint = utils_misc.get_path(test.debugdir, host_mountpoint)
+        file_chk_for_win = params.get(
+            "coredump_check_win",
+            file_check_win_default)
+        file_chk_for_linux = params.get("coredump_check_linux",
+                                        file_check_linux_default)
+
+        # check if the host_mountpoint exists.
+        if not (os.path.isdir(host_mountpoint) and
+                os.path.exists(host_mountpoint)):
+            os.makedirs(host_mountpoint)
+
+        coredump_file_exists = False
+        check_files = [file_chk_for_win, file_chk_for_linux]
+        check_results = []
+
+        error.context("Get all the images name", logging.info)
+        images = get_images()
+        error.context("images: %s" % images, logging.info)
+
+        # find all the images
+        # mount per-image to check if the dump file exists
+        error.context("Check coredump file per-image", logging.info)
+        for image in images:
+            status, chk_msgs = check_images_coredump(image,
+                                                     host_mountpoint,
+                                                     check_files,
+                                                     test.debugdir)
+            coredump_file_exists = coredump_file_exists or status
+            if status:
+                check_results.append((image, chk_msgs))
+
+        # if found, report the result
+        if coredump_file_exists:
+            report_msg = format_report(check_results)
+            msgbox.append(report_msg)
+    except ValueError, err:
+        warn_msg = "%s: Need packages: " % str(err)
+        warn_msg += "libguestfs libguestfs-tools bguestfs-winsupport"
         raise error.TestWarn(warn_msg)
-
```

Comment:
hi,xu
The test result is GOOD. 
virt.qemu.smp_16.4096m.repeat1.post_check.Host_RHEL.7.2.raw.virtio_blk.up.virtio_net.RHEL.7.1.x86_64.io-github-autotest-qemu.check_coredump    GOOD       14       completed successfully

But check the debug log,found image is mounted fail since lack of packages. So i think this case's result should be ERROR or WARN.

06/09 16:18:34 DEBUG|utils_libg:0080| stderr: libguestfs: error: appliance closed the connection unexpectedly.
This usually means the libguestfs appliance crashed.
See http://libguestfs.org/guestfs-faq.1.html#debugging-libguestfs
for information about how to debug libguestfs and report bugs.
libguestfs: error: guestfs_launch failed.
This usually means the libguestfs appliance failed to start or crashed.
See http://libguestfs.org/guestfs-faq.1.html#debugging-libguestfs
or run 'libguestfs-test-tool' and post the *complete* output into a
bug report or message to the libguestfs mailing list.
06/09 16:18:34 ERROR|     error:0085| Context: Could not mount guest image /home/my_auto/autotest-devel/client/tests/virt/shared/data/images/RHEL-Server-7.1-64-virtio.raw